packages/next/src/server/request/connection.ts

```diff
@@ -44,15 +44,15 @@ export function connection(): Promise<void> {
 
     if (workStore.dynamicShouldError) {
       throw new StaticGenBailoutError(
-        `Route ${workStore.route} with \`dynamic = "error"\` couldn't be rendered statically because it used \`connection\`. See more info here: https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic#dynamic-rendering`
+        `Route ${workStore.route} with \`dynamic = "error"\` couldn't be rendered statically because it used \`connection()\`. \`connection()\` is a Promise and must be unwrapped with \`await\` or \`React.use()\` before accessing its properties. See more info here: https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic#dynamic-rendering`
```

Comment:
This one shouldn't be updated. this error only happens in non dynamicIO cases where simply calling connection is like marking the render as dynamic. Additionally while connection returns a promise it has no properties so it's only used when you want to mark the current scope as dynamic. I think you can revert the wording and just add `()` to the function expression